components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -350,6 +350,18 @@ func generateStatefulSet(instance *v1beta1.Notebook) *appsv1.StatefulSet {
 		Value: "/notebook/" + instance.Namespace + "/" + instance.Name,
 	})
 
+	// Add readiness probe
+	if container.ReadinessProbe == nil {
```

Comment:
This is just to prevent overwriting an existing probe right?

Reply:
Yup, it was especially useful for testing.  Most if not all users create notebooks through the UI. 
If they created a notebook manually and added a readiness probe, they probably know what they're doing.